diff --git a/docker/local.dockerfile b/docker/local.dockerfile
index 541e387..8e22afb 100644
--- a/docker/local.dockerfile
+++ b/docker/local.dockerfile
@@ -1,6 +1,8 @@
-FROM golang:1.22-alpine as builder
+FROM lcr.loongnix.cn/library/debian:sid as builder
 
-ARG VERSION=unknown
+RUN apt update && apt install -y golang ca-certificates
+
+ARG VERSION=0.0.4
 
 # copy project
 COPY . /app
@@ -9,7 +11,7 @@ COPY . /app
 WORKDIR /app
 
 # using goproxy if you have network issues
-# ENV GOPROXY=https://goproxy.cn,direct
+RUN GO111MODULE=on GOPROXY=https://goproxy.cn go mod vendor
 
 # build
 RUN go build \
@@ -22,7 +24,7 @@ RUN go build \
 COPY entrypoint.sh /app/entrypoint.sh
 RUN chmod +x /app/entrypoint.sh
 
-FROM ubuntu:24.04
+FROM lcr.loongnix.cn/library/debian:sid
 
 COPY --from=builder /app/main /app/main
 COPY --from=builder /app/entrypoint.sh /app/entrypoint.sh
@@ -33,7 +35,7 @@ WORKDIR /app
 ARG PLATFORM=local
 
 # Install python3.12 if PLATFORM is local
-RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y curl python3.12 python3.12-venv python3.12-dev python3-pip ffmpeg build-essential \
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y curl python3.12 python3.12-venv python3.12-dev python3-pip ffmpeg build-essential rustc cargo \
     && apt-get clean \
     && rm -rf /var/lib/apt/lists/* \
     && update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.12 1;
