From 7297cb8e0e0d01c8b227718495cd576feba48dc7 Mon Sep 17 00:00:00 2001
From: Huang Yang <huangyang@loongson.cn>
Date: Mon, 30 Jun 2025 10:22:24 +0800
Subject: [PATCH] pvc-autoresizer add loong64

---
 Dockerfile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Dockerfile b/Dockerfile
index 3bc1bfa..08e40e8 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,5 +1,5 @@
 # Stage1: Build the pvc-autoresizer binary
-FROM golang:1.19 as builder
+FROM lcr.loongnix.cn/library/golang:1.24 as builder
 
 WORKDIR /workspace
 
-- 
2.41.0

