diff --git a/Zend/zend_long.h b/Zend/zend_long.h
index 453098c1..830c7e72 100644
--- a/Zend/zend_long.h
+++ b/Zend/zend_long.h
@@ -22,7 +22,7 @@
 #include "main/php_stdint.h"
 
 /* This is the heart of the whole int64 enablement in zval. */
-#if defined(__x86_64__) || defined(__LP64__) || defined(_LP64) || defined(_WIN64)
+#if defined(__x86_64__) || defined(__LP64__) || defined(_LP64) || defined(_WIN64) || defined(__loongarch64)
 # define ZEND_ENABLE_ZVAL_LONG64 1
 #endif
 
diff --git a/main/streams/cast.c b/main/streams/cast.c
index db0f039e..80034e1e 100644
--- a/main/streams/cast.c
+++ b/main/streams/cast.c
@@ -70,10 +70,10 @@ static int stream_cookie_writer(void *cookie, const char *buffer, int size)
 	return php_stream_write((php_stream *)cookie, (char *)buffer, size);
 }
 
-static PHP_FPOS_T stream_cookie_seeker(void *cookie, zend_off_t position, int whence)
+static PHP_FPOS_T stream_cookie_seeker(void *cookie, PHP_FPOS_T position, int whence)
 {
 
-	return (PHP_FPOS_T)php_stream_seek((php_stream *)cookie, position, whence);
+	return (PHP_FPOS_T)php_stream_seek((php_stream *)cookie, (zend_off_t)position, whence);
 }
 
 static int stream_cookie_closer(void *cookie)
