From 6fb528af49b0a1ebcf3526747ceb7096aa6a1055 Mon Sep 17 00:00:00 2001
From: Huang Yang <huangyang@loongson.cn>
Date: Mon, 20 Apr 2026 17:02:12 +0800
Subject: [PATCH] 2FAuth 6.1.1 add loongarch support

---
 Dockerfile           | 21 +++++++++++----------
 docker/entrypoint.sh |  4 ++--
 2 files changed, 13 insertions(+), 12 deletions(-)

diff --git a/Dockerfile b/Dockerfile
index a95ab7a..0b3f0e1 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,18 +1,18 @@
-ARG BUILDPLATFORM=linux/amd64
+ARG BUILDPLATFORM=linux/loong64
 ARG TARGETPLATFORM
 ARG ALPINE_VERSION=3.23
-ARG PHP_VERSION=8.4-alpine${ALPINE_VERSION}
-ARG COMPOSER_VERSION=2.9
+ARG PHP_VERSION=8.4.15-cli-alpine${ALPINE_VERSION}
+ARG COMPOSER_VERSION=v2.9.5
 ARG SUPERVISORD_VERSION=v0.7.3
 
 ARG UID=1000
 ARG GID=1000
 
-FROM --platform=${BUILDPLATFORM} composer:${COMPOSER_VERSION} AS build-composer
-FROM composer:${COMPOSER_VERSION} AS composer
-FROM qmcgaw/binpot:supervisord-${SUPERVISORD_VERSION} AS supervisord
+FROM --platform=${BUILDPLATFORM} lcr.loongnix.cn/library/composer:${COMPOSER_VERSION} AS build-composer
+FROM lcr.loongnix.cn/library/composer:${COMPOSER_VERSION} AS composer
+#FROM qmcgaw/binpot:supervisord-${SUPERVISORD_VERSION} AS supervisord
 
-FROM --platform=${BUILDPLATFORM} php:${PHP_VERSION} AS vendor
+FROM --platform=${BUILDPLATFORM} lcr.loongnix.cn/library/php:${PHP_VERSION} AS vendor
 ARG UID=1000
 ARG GID=1000
 COPY --from=build-composer --chown=${UID}:${GID} /usr/bin/composer /usr/bin/composer
@@ -41,7 +41,7 @@ ARG GID=1000
 # Composer 2
 COPY --from=composer --chown=${UID}:${GID} /usr/bin/composer /usr/bin/composer
 # Supervisord from https://github.com/ochinchina/supervisord
-COPY --from=supervisord --chown=${UID}:${GID} /bin /usr/local/bin/supervisord
+# COPY --from=supervisord --chown=${UID}:${GID} /bin /usr/local/bin/supervisord
 
 # Install PHP and PHP system dependencies
 RUN apk add --update --no-cache \
@@ -56,7 +56,8 @@ RUN apk add --update --no-cache \
     # Runtime dependencies
     php84-session php84-openssl \
     # Nginx and PHP FPM to serve over HTTP
-    php84-fpm nginx
+    php84-fpm nginx \
+    supervisor
 
 # PHP FPM configuration
 # Change username and ownership in php-fpm pool config
@@ -81,7 +82,7 @@ COPY --chown=${UID}:${GID} docker/nginx.conf /etc/nginx/nginx.conf
 RUN nginx -t
 
 # Supervisord configuration
-COPY --chown=${UID}:${GID} docker/supervisord.conf /etc/supervisor/supervisord.conf
+COPY --chown=${UID}:${GID} docker/supervisord.conf /etc/supervisord.conf
 
 # Create end user directory
 RUN mkdir -p /2fauth && \
diff --git a/docker/entrypoint.sh b/docker/entrypoint.sh
index ff4c116..810fbac 100644
--- a/docker/entrypoint.sh
+++ b/docker/entrypoint.sh
@@ -37,7 +37,7 @@ file_env() {
 echo "Running version ${VERSION} commit ${COMMIT} built on ${CREATED}"
 
 # Show versions
-echo "supervisord version: $(supervisord version)"
+echo "supervisord version: $(supervisord -v 2>&1)"
 php-fpm84 -v | head -n 1
 nginx -v
 
@@ -119,4 +119,4 @@ php artisan clear-compiled
 # Clearing and Caching config, events, routes, views
 php artisan optimize
 
-supervisord
+exec supervisord -c /etc/supervisord.conf
-- 
2.47.3

